File: level_1/linklist_alter/langs/compiler.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* insert(struct Node* head, int value) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->next = NULL;

    if (head == NULL) {
        return newNode;
    }

    struct Node* current = head;
    while (current->next != NULL) {
        current = current->next;
    }

    current->next = newNode;
    return head;
}

// Function to print the linked list
void printList(struct Node* head) {
    while (head != NULL) {
        printf("%d ", head->data);
        head = head->next;
    }
    printf("\n");
}

int main() {
    int N, value;

    scanf("%d", &N);
    struct Node* head = NULL;

    for (int i = 0; i < N; ++i) {
        scanf("%d", &value);
        head = insert(head, value);
    }

    struct Node** result = splitAlternateNodes(head);

    printList(result[0]);
    printList(result[1]);

    free(result[0]);
    free(result[1]);

    return 0;
}

File: level_1/linklist_alter/langs/compiler.java
class Node {
    int data;
    Node next;

    Node(int val) {
        data = val;
        next = null;
    }
}

class Main {
    public static Node insert(Node head, int value) {
        if (head == null) {
            return new Node(value);
        }
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }
        current.next = new Node(value);
        return head;
    }

    public static void printList(Node head) {
        while (head != null) {
            System.out.print(head.data + " ");
            head = head.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int N = scanner.nextInt();
        Node head = null;
        for (int i = 0; i < N; ++i) {
            int value = scanner.nextInt();
            head = insert(head, value);
        }
        Test test = new Test();
        ArrayList<Node> result = test.splitAlternateNodes(head);
        printList(result.get(0));
        printList(result.get(1));
    }
}
